lib/stream.js

```diff
@@ -90,43 +93,55 @@ const throwOnStreamError = async stream => {
 // The streams created by the `std*` options are automatically ended by `.pipe()`.
 // However `.pipe()` only does so when the source stream ended, not when it errored.
 // Therefore, when `childProcess.stdin|stdout|stderr` errors, those streams must be manually destroyed.
-const cleanupStdioStreams = (stdioStreams, error) => {
-	for (const stdioStream of stdioStreams) {
-		if (stdioStream.autoDestroy) {
-			stdioStream.value.destroy(error);
+const cleanupStdioStreams = (customStreams, error) => {
+	for (const customStream of customStreams) {
+		if (customStream.autoDestroy) {
+			customStream.value.destroy(error);
 		}
 	}
 };
 
 // Retrieve result of child process: exit code, signal, error, streams (stdout/stderr/all)
 export const getSpawnedResult = async (
 	spawned,
-	{encoding, buffer, maxBuffer},
+	{encoding, buffer, maxBuffer, timeout, killSignal, cleanup, detached},
+	context,
 	stdioStreams,
-	processDone,
 ) => {
+	const finalizers = [];
```

Comment:
Array of functions to call after completion (success or error). At the moment, this cleans up the `timeout` and `cleanup` option listeners.